colorbleed/maya/customize.py

```diff
@@ -0,0 +1,66 @@
+"""A set of commands that install overrides to Maya's UI"""
+
+import maya.cmds as mc
+import maya.mel as mel
+from functools import partial
+import logging
+
+
+log = logging.getLogger(__name__)
+
+COMPONENT_MASK_ORIGINAL = {}
+
+
+def override_component_mask_commands():
+    """Override component mask ctrl+click behavior.
+
+    This implements special behavior for Maya's component
+    mask menu items where a ctrl+click will instantly make
+    it an isolated behavior disabling all others.
+    
+    Tested in Maya 2016 and 2018
+
+    """
+    log.info("Installing override_component_mask_commands..")
+
+    # Get all object mask buttons
+    buttons = mc.formLayout("objectMaskIcons",
+                            query=True,
+                            childArray=True)
+    # Skip the triangle list item
+    buttons = [btn for btn in buttons if btn != "objPickMenuLayout"]
+
+    def _on_changed_callback(raw_command, state):
```

Comment:
This is a private function AND a protected function. Use either a private function or protected function.
I would stay away from nesting functions in functions because, IMO, it makes the code less clear.

Reply:
It's nested because it's local callback. I don't want it globally. It should stay there.

Reply:
We could remove the underscore though if that makes it more readable. :)

Reply:
I think your argument for nesting it could have been better. After a minor google search I found this:
> A closure simply causes the inner function to remember the state of its environment when called. Beginners often think that a closure is the inner function, when it’s really caused by the inner function. The closure “closes” the local variable on the stack and this stays around after the the stack creation has finished executing.

If this effect is why you use it I am fully behind using it as a *nested function* but if not I would recommend moving it and set it private.

Reply:
Yup, it also uses the parent function's `buttons` list, which obviously is another reason. Good point. I'll remove the underscore and we can merge!